yb-voyager/src/anon/sql_anonymizer.go

```diff
@@ -1056,6 +1219,87 @@ func (a *SqlAnonymizer) handleIndexObjectNodes(msg protoreflect.Message) (err er
 			return fmt.Errorf("anon index column alias: %w", err)
 		}
 
+		// 4) Operator class: (Opclass != nil)
+		//    CREATE INDEX ... ON tbl(col opclass_schema.opclass_name(opts));
+		if ie.Opclass != nil {
+			err = a.anonymizeIndexOpclass(ie.Opclass)
+			if err != nil {
+				return fmt.Errorf("anon index opclass: %w", err)
+			}
+		}
+
+		// 5) Operator class options: (Opclassopts != nil)
+		//    CREATE INDEX ... ON tbl(col opclass_name(siglen='32'));
+		if ie.Opclassopts != nil {
+			err = a.anonymizeIndexOpclassOptions(ie.Opclassopts)
+			if err != nil {
+				return fmt.Errorf("anon index opclass options: %w", err)
+			}
+		}
+
+	}
+
+	return nil
+}
+
+// anonymizeIndexOpclass anonymizes operator class names in index definitions
+func (a *SqlAnonymizer) anonymizeIndexOpclass(opclass []*pg_query.Node) error {
+	if opclass == nil {
+		return nil
+	}
+
+	// Handle operator class names (usually 1-2 elements: [schema, opclass_name])
+	for _, op := range opclass {
+		if op == nil {
+			continue
+		}
+
+		// Extract string value from the operator class node
+		if strNode := op.GetString_(); strNode != nil {
+			hashedName, err := a.registry.GetHash(OPCLASS_KIND_PREFIX, strNode.Sval)
+			if err != nil {
+				return fmt.Errorf("anon opclass name: %w", err)
+			}
+			strNode.Sval = hashedName
+		}
+	}
+
+	return nil
+}
+
+// anonymizeIndexOpclassOptions anonymizes operator class options in index definitions
+func (a *SqlAnonymizer) anonymizeIndexOpclassOptions(opts []*pg_query.Node) error {
+	if opts == nil {
+		return nil
+	}
+
+	for _, opt := range opts {
+		if opt == nil {
+			continue
+		}
+
+		// Handle DefElem nodes (e.g., siglen='32')
+		if defElem := opt.GetDefElem(); defElem != nil {
+			// Anonymize the option name if needed
+			if defElem.Defname != "" {
+				hashedName, err := a.registry.GetHash(PARAMETER_KIND_PREFIX, defElem.Defname)
+				if err != nil {
+					return fmt.Errorf("anon opclass option name: %w", err)
+				}
+				defElem.Defname = hashedName
+			}
+
+			// Handle the argument value (e.g., '32')
+			if defElem.Arg != nil {
+				if strNode := defElem.Arg.GetString_(); strNode != nil {
```

Comment:
can this arg only have string values?
AFAIU, this `Arg` is a generic Node which can have anything like a typename, constant, function, or expression, and another Arg again

Reply:
That won't be feasible to determine.
I think best feasible option is to just consider that the function arguments here as constant.

Let me know what do you think.

Reply:
It might be possible to iterate over all possible cases we can figure out, for reference, some kind this I did for one of the issue check `DoesNodeHandleJsonbData` function in helper_struct.go

But yeah might not be a common case for this particular DDL type, so probably we can come back to it if we find a case (most likely with DMLs we have to do that)